emails/views.py

```diff
@@ -479,7 +480,18 @@ def _handle_received(message_json: AWS_SNSMessageJSON) -> HttpResponse:
         return HttpResponse("Address does not exist", status=404)
 
     _record_receipt_verdicts(receipt, "relay_recipient")
-    from_address = parseaddr(common_headers["from"][0])[1]
+    from_addresses = parse_email_header(common_headers["from"][0])
```

Comment:
question (non-blocking): should we also change the `parseaddr` calls in `_handle_bounce` and `_handle_complaint` to call `parse_email_header` ? (We're also using `parseaddr` in `_get_relay_recipient_from_message_json`, but since those are our own email addresses, we seem to send all those with well-formed addresses.

Reply:
maybe yes, and the code that parses the `To:` and `CC:` fields as well. This may be bigger than a "quick tech-debt" issue.

Reply:
I'm still happy to merge this as an incremental improvement on what we have now.